scripts/xref-check.escript

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env escript
+
+%%% @doc
+%%% Unlike rebar3 xref check, this script runs the full xref checks in the EMQX release dir,
+%%% meaning all the modules for release are analysed.
+%%% For behavior configuration, all rebar3 related modules attributes, filters are not used in this script,
+%%% instead all the filters, checks are defined in `xref_check.eterm`
+main(_) ->
+    {ok, [Jobs]} = file:consult("scripts/xref_check.eterm"),
+    lists:foreach(fun(#{ name := Name
+                       , analysis := Analysis
+                       , excl_apps := ExclApps
+                       , excl_mods := ExclMods
+                       , filters := Filters
+                       }) ->
+                          xref:start(Name),
+                          Tid = ets:new(Name, [ordered_set, named_table]),
+                          xref:set_default(Name, [{verbose,false}, {warnings,false}]),
+                          Profile = case filelib:is_file("EMQX_ENTERPRISE") of
+                                        true -> emqx-ee;
+                                        false -> emqx
+                                    end,
+                          Dir = filename:join(["_build/",  Profile, "rel/emqx/lib/"]),
+                          xref:add_release(Name, Dir),
+                          xref:add_application(Name, code:lib_dir(erts)),
+                          [ case xref:remove_application(Name, App) of
+                                ok -> ok;
+                                {error, xref_base, {no_such_application, _}} -> ok
+                            end || App <- ExclApps
+                          ],
+
+                          [case xref:remove_module(Name, M) of
+                               ok -> ok;
+                               %% but in doc it should return '{error, module(), Reason}`
+                               {error, xref_base, {no_such_module, M}} -> ok
+                           end || M <- ExclMods
+                          ],
+                          ModuleInfos = xref:info(Name, modules),
+                          LibInfos = xref:info(Name, modules),
+                          true = ets:insert(Tid, ModuleInfos ++ LibInfos),
+                          {ok, Res0} = xref:analyse(Name, Analysis),
+                          Res = Res0 -- Filters,
+                          Res =/= [] andalso
+                              begin
+                                  put(is_warn_found, true),
+                                  io:format("** Warnings for ~p~n : ~p~n", [Name, Res])
```

Comment:
Nit: maybe print the results in a new line, so the results are wider?

Reply:
It is in new line already. 
The output is already readable to me.

example: 
```
===> Running cross reference analysis...
** Warnings for undefined_functions
 : [{'Elixir.Poison','decode!',1},
    {'Elixir.Poison','encode!',1},
    {'Elixir.Poison.EncodeError',exception,1},
    {'Elixir.Poison.Encoder.Atom',encode,2},
    {'Elixir.Poison.Encoder.BitString',encode,2},
    {'Elixir.Poison.Encoder.Float',encode,2},
    {'Elixir.Poison.Encoder.Integer',encode,2}]
make: *** [Makefile:119: xref] Error 1
```

Reply:
Ooops, missed that newline, ignore me.